poc_iot_injector/src/server.rs

```diff
@@ -48,171 +40,96 @@ impl Server {
     pub async fn new(settings: &Settings) -> Result<Self, NewServerError> {
         let pool = settings.database.connect(10).await?;
         let keypair = settings.keypair()?;
-        let tick_time = settings.trigger_interval();
-        let submission_offset = settings.submission_offset();
-        let max_lookback_age = settings.max_lookback_age();
-        // Check meta for last_poc_submission_ts, if not found, use the env var and insert it
-        let last_poc_submission_ts =
-            MetaValue::<i64>::fetch_or_insert_with(&pool, "last_reward_end_time", || {
-                settings.last_poc_submission
-            })
-            .await?;
 
         let result = Self {
-            pool: pool.clone(),
+            pool,
             settings: settings.clone(),
             keypair: Arc::new(keypair),
-            tick_time,
-            submission_offset,
             // Only create txn_service if do_submission is true
             txn_service: settings
                 .do_submission
                 .then_some(TransactionService::from_settings(&settings.transactions)),
-            iot_verifier_store: FileStore::from_settings(&settings.verifier).await?,
-            last_poc_submission_ts,
-            max_lookback_age,
         };
         Ok(result)
     }
 
-    pub async fn run(&mut self, shutdown: &triggered::Listener) -> anyhow::Result<()> {
+    pub async fn run(
+        &mut self,
+        shutdown: &triggered::Listener,
+        mut receiver: Receiver<FileInfoStream<IotPoc>>,
+    ) -> anyhow::Result<()> {
         tracing::info!("starting poc-iot-injector server");
-        let mut poc_iot_timer = time::interval(self.tick_time);
-        poc_iot_timer.set_missed_tick_behavior(time::MissedTickBehavior::Delay);
-
         loop {
-            if shutdown.is_triggered() {
-                break;
-            }
             tokio::select! {
                 _ = shutdown.clone() => break,
-                _ = poc_iot_timer.tick() => match self.handle_poc_tick().await {
-                    Ok(()) => (),
-                    Err(err) => {
-                        tracing::error!("fatal poc_iot_injector error: {err:?}");
-                        return Err(err)
-                    }
+                msg = receiver.recv() => if let Some(stream) =  msg {
```

Comment:
this seems weird in that why would the received msg potentially be a None ? why would the filestore push a None stream

Reply:
https://docs.rs/tokio/1.25.0/tokio/sync/mpsc/struct.Receiver.html#method.recv

The `recv` function always returns an `Option`. The file_store is not sending an Option on the channel.

Reply:
To further elaborate, recv will return a None if the Sender has been dropped